<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Q&A 5 How do you tidy the genotype matrix from a .ped file in R? | gwas.knit</title>
  <meta name="description" content="Learning GWAS One Step at a Time with R" />
  <meta name="generator" content="bookdown 0.42 and GitBook 2.6.7" />

  <meta property="og:title" content="Q&A 5 How do you tidy the genotype matrix from a .ped file in R? | gwas.knit" />
  <meta property="og:type" content="book" />
  
  
  <meta name="github-repo" content="https://github.com/datainsights/cdi-gwas/" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Q&A 5 How do you tidy the genotype matrix from a .ped file in R? | gwas.knit" />
  
  
  




  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  <link rel="shortcut icon" href="favicon.ico" type="image/x-icon" />
<link rel="prev" href="how-do-you-inspect-the-structure-and-contents-of-gwas-input-files-in-r.html"/>
<link rel="next" href="how-do-you-recode-allele-strings-into-numeric-count-format-for-gwas.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { display: inline-block; text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
  
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="https://datainsights.github.io/cdi-gwas">Genome Wide Studies</a></li>

<li class="divider"></li>
<li><a href="index.html#section" id="toc-section"></a></li>
<li class="chapter" data-level="" data-path="welcome-to-cdi-gwas-for-population-health.html"><a href="welcome-to-cdi-gwas-for-population-health.html"><i class="fa fa-check"></i>Welcome to CDI ‚Äì GWAS for Population Health</a>
<ul>
<li class="chapter" data-level="" data-path="welcome-to-cdi-gwas-for-population-health.html"><a href="welcome-to-cdi-gwas-for-population-health.html#the-cdi-learning-path"><i class="fa fa-check"></i>üåê The CDI Learning Path</a></li>
</ul></li>
<li class="part"><span><b>I DATA EXPLORATION</b></span></li>
<li class="chapter" data-level="1" data-path="how-do-you-create-a-gwas-project-directory-ready-for-analysis.html"><a href="how-do-you-create-a-gwas-project-directory-ready-for-analysis.html"><i class="fa fa-check"></i><b>1</b> How do you create a GWAS project directory ready for analysis?</a>
<ul>
<li class="chapter" data-level="1.1" data-path="how-do-you-create-a-gwas-project-directory-ready-for-analysis.html"><a href="how-do-you-create-a-gwas-project-directory-ready-for-analysis.html#explanation"><i class="fa fa-check"></i><b>1.1</b> Explanation</a></li>
<li class="chapter" data-level="1.2" data-path="how-do-you-create-a-gwas-project-directory-ready-for-analysis.html"><a href="how-do-you-create-a-gwas-project-directory-ready-for-analysis.html#bash-terminal"><i class="fa fa-check"></i><b>1.2</b> Bash (Terminal)</a></li>
<li class="chapter" data-level="1.3" data-path="how-do-you-create-a-gwas-project-directory-ready-for-analysis.html"><a href="how-do-you-create-a-gwas-project-directory-ready-for-analysis.html#python-code"><i class="fa fa-check"></i><b>1.3</b> Python Code</a></li>
<li class="chapter" data-level="1.4" data-path="how-do-you-create-a-gwas-project-directory-ready-for-analysis.html"><a href="how-do-you-create-a-gwas-project-directory-ready-for-analysis.html#r-code"><i class="fa fa-check"></i><b>1.4</b> R Code</a></li>
<li class="chapter" data-level="1.5" data-path="how-do-you-create-a-gwas-project-directory-ready-for-analysis.html"><a href="how-do-you-create-a-gwas-project-directory-ready-for-analysis.html#import-libraries"><i class="fa fa-check"></i><b>1.5</b> Import libraries</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="how-do-you-prepare-a-public-gwas-dataset-for-r-based-analysis.html"><a href="how-do-you-prepare-a-public-gwas-dataset-for-r-based-analysis.html"><i class="fa fa-check"></i><b>2</b> How do you prepare a public GWAS dataset for R-based analysis?</a>
<ul>
<li class="chapter" data-level="2.1" data-path="how-do-you-prepare-a-public-gwas-dataset-for-r-based-analysis.html"><a href="how-do-you-prepare-a-public-gwas-dataset-for-r-based-analysis.html#explanation-1"><i class="fa fa-check"></i><b>2.1</b> Explanation</a></li>
<li class="chapter" data-level="2.2" data-path="how-do-you-prepare-a-public-gwas-dataset-for-r-based-analysis.html"><a href="how-do-you-prepare-a-public-gwas-dataset-for-r-based-analysis.html#bash-script"><i class="fa fa-check"></i><b>2.2</b> Bash Script</a></li>
<li class="chapter" data-level="2.3" data-path="how-do-you-prepare-a-public-gwas-dataset-for-r-based-analysis.html"><a href="how-do-you-prepare-a-public-gwas-dataset-for-r-based-analysis.html#file-structure"><i class="fa fa-check"></i><b>2.3</b> File Structure</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="how-do-you-efficiently-load-and-store-gwas-data-files-in-r.html"><a href="how-do-you-efficiently-load-and-store-gwas-data-files-in-r.html"><i class="fa fa-check"></i><b>3</b> How do you efficiently load and store GWAS data files in R?</a>
<ul>
<li class="chapter" data-level="3.1" data-path="how-do-you-efficiently-load-and-store-gwas-data-files-in-r.html"><a href="how-do-you-efficiently-load-and-store-gwas-data-files-in-r.html#explanation-2"><i class="fa fa-check"></i><b>3.1</b> Explanation</a></li>
<li class="chapter" data-level="3.2" data-path="how-do-you-efficiently-load-and-store-gwas-data-files-in-r.html"><a href="how-do-you-efficiently-load-and-store-gwas-data-files-in-r.html#r-code-1"><i class="fa fa-check"></i><b>3.2</b> R Code</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="how-do-you-inspect-the-structure-and-contents-of-gwas-input-files-in-r.html"><a href="how-do-you-inspect-the-structure-and-contents-of-gwas-input-files-in-r.html"><i class="fa fa-check"></i><b>4</b> How do you inspect the structure and contents of GWAS input files in R?</a>
<ul>
<li class="chapter" data-level="4.1" data-path="how-do-you-inspect-the-structure-and-contents-of-gwas-input-files-in-r.html"><a href="how-do-you-inspect-the-structure-and-contents-of-gwas-input-files-in-r.html#explanation-3"><i class="fa fa-check"></i><b>4.1</b> Explanation</a></li>
<li class="chapter" data-level="4.2" data-path="how-do-you-inspect-the-structure-and-contents-of-gwas-input-files-in-r.html"><a href="how-do-you-inspect-the-structure-and-contents-of-gwas-input-files-in-r.html#r-code-2"><i class="fa fa-check"></i><b>4.2</b> R Code</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="how-do-you-tidy-the-genotype-matrix-from-a-.ped-file-in-r.html"><a href="how-do-you-tidy-the-genotype-matrix-from-a-.ped-file-in-r.html"><i class="fa fa-check"></i><b>5</b> How do you tidy the genotype matrix from a <code>.ped</code> file in R?</a>
<ul>
<li class="chapter" data-level="5.1" data-path="how-do-you-tidy-the-genotype-matrix-from-a-.ped-file-in-r.html"><a href="how-do-you-tidy-the-genotype-matrix-from-a-.ped-file-in-r.html#explanation-4"><i class="fa fa-check"></i><b>5.1</b> Explanation</a></li>
<li class="chapter" data-level="5.2" data-path="how-do-you-tidy-the-genotype-matrix-from-a-.ped-file-in-r.html"><a href="how-do-you-tidy-the-genotype-matrix-from-a-.ped-file-in-r.html#r-code-3"><i class="fa fa-check"></i><b>5.2</b> R Code</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="how-do-you-recode-allele-strings-into-numeric-count-format-for-gwas.html"><a href="how-do-you-recode-allele-strings-into-numeric-count-format-for-gwas.html"><i class="fa fa-check"></i><b>6</b> How do you recode allele strings into numeric count format for GWAS?</a>
<ul>
<li class="chapter" data-level="6.1" data-path="how-do-you-recode-allele-strings-into-numeric-count-format-for-gwas.html"><a href="how-do-you-recode-allele-strings-into-numeric-count-format-for-gwas.html#explanation-5"><i class="fa fa-check"></i><b>6.1</b> Explanation</a></li>
<li class="chapter" data-level="6.2" data-path="how-do-you-recode-allele-strings-into-numeric-count-format-for-gwas.html"><a href="how-do-you-recode-allele-strings-into-numeric-count-format-for-gwas.html#r-code-4"><i class="fa fa-check"></i><b>6.2</b> R Code</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="how-do-you-filter-snps-and-samples-based-on-missing-data-and-minor-allele-frequency.html"><a href="how-do-you-filter-snps-and-samples-based-on-missing-data-and-minor-allele-frequency.html"><i class="fa fa-check"></i><b>7</b> How do you filter SNPs and samples based on missing data and minor allele frequency?</a>
<ul>
<li class="chapter" data-level="7.1" data-path="how-do-you-filter-snps-and-samples-based-on-missing-data-and-minor-allele-frequency.html"><a href="how-do-you-filter-snps-and-samples-based-on-missing-data-and-minor-allele-frequency.html#explanation-6"><i class="fa fa-check"></i><b>7.1</b> Explanation</a></li>
<li class="chapter" data-level="7.2" data-path="how-do-you-filter-snps-and-samples-based-on-missing-data-and-minor-allele-frequency.html"><a href="how-do-you-filter-snps-and-samples-based-on-missing-data-and-minor-allele-frequency.html#r-code-5"><i class="fa fa-check"></i><b>7.2</b> R Code</a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="how-do-you-impute-missing-genotype-values-before-gwas-analysis.html"><a href="how-do-you-impute-missing-genotype-values-before-gwas-analysis.html"><i class="fa fa-check"></i><b>8</b> How do you impute missing genotype values before GWAS analysis?</a>
<ul>
<li class="chapter" data-level="8.1" data-path="how-do-you-impute-missing-genotype-values-before-gwas-analysis.html"><a href="how-do-you-impute-missing-genotype-values-before-gwas-analysis.html#explanation-7"><i class="fa fa-check"></i><b>8.1</b> Explanation</a></li>
<li class="chapter" data-level="8.2" data-path="how-do-you-impute-missing-genotype-values-before-gwas-analysis.html"><a href="how-do-you-impute-missing-genotype-values-before-gwas-analysis.html#r-code-6"><i class="fa fa-check"></i><b>8.2</b> R Code</a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="how-do-you-perform-pca-on-genotype-data-to-assess-population-structure.html"><a href="how-do-you-perform-pca-on-genotype-data-to-assess-population-structure.html"><i class="fa fa-check"></i><b>9</b> How do you perform PCA on genotype data to assess population structure?</a>
<ul>
<li class="chapter" data-level="9.1" data-path="how-do-you-perform-pca-on-genotype-data-to-assess-population-structure.html"><a href="how-do-you-perform-pca-on-genotype-data-to-assess-population-structure.html#explanation-8"><i class="fa fa-check"></i><b>9.1</b> Explanation</a></li>
<li class="chapter" data-level="9.2" data-path="how-do-you-perform-pca-on-genotype-data-to-assess-population-structure.html"><a href="how-do-you-perform-pca-on-genotype-data-to-assess-population-structure.html#r-code-7"><i class="fa fa-check"></i><b>9.2</b> R Code</a></li>
</ul></li>
<li class="chapter" data-level="10" data-path="how-do-you-include-pca-covariates-in-a-gwas-model.html"><a href="how-do-you-include-pca-covariates-in-a-gwas-model.html"><i class="fa fa-check"></i><b>10</b> How do you include PCA covariates in a GWAS model?</a>
<ul>
<li class="chapter" data-level="10.1" data-path="how-do-you-include-pca-covariates-in-a-gwas-model.html"><a href="how-do-you-include-pca-covariates-in-a-gwas-model.html#explanation-9"><i class="fa fa-check"></i><b>10.1</b> Explanation</a></li>
<li class="chapter" data-level="10.2" data-path="how-do-you-include-pca-covariates-in-a-gwas-model.html"><a href="how-do-you-include-pca-covariates-in-a-gwas-model.html#r-code-8"><i class="fa fa-check"></i><b>10.2</b> R Code</a></li>
</ul></li>
<li class="chapter" data-level="11" data-path="how-do-you-interpret-gwas-model-results-with-pca-covariates.html"><a href="how-do-you-interpret-gwas-model-results-with-pca-covariates.html"><i class="fa fa-check"></i><b>11</b> How do you interpret GWAS model results with PCA covariates?</a>
<ul>
<li class="chapter" data-level="11.1" data-path="how-do-you-interpret-gwas-model-results-with-pca-covariates.html"><a href="how-do-you-interpret-gwas-model-results-with-pca-covariates.html#explanation-10"><i class="fa fa-check"></i><b>11.1</b> Explanation</a></li>
<li class="chapter" data-level="11.2" data-path="how-do-you-interpret-gwas-model-results-with-pca-covariates.html"><a href="how-do-you-interpret-gwas-model-results-with-pca-covariates.html#r-model-output-summary"><i class="fa fa-check"></i><b>11.2</b> R Model Output Summary</a></li>
</ul></li>
<li class="part"><span><b>II GWAS ANALYSIS &amp; VIZ</b></span></li>
<li class="chapter" data-level="12" data-path="how-do-you-perform-a-genome-wide-snp-scan-to-generate-gwas-results.html"><a href="how-do-you-perform-a-genome-wide-snp-scan-to-generate-gwas-results.html"><i class="fa fa-check"></i><b>12</b> How do you perform a genome-wide SNP scan to generate GWAS results?</a>
<ul>
<li class="chapter" data-level="12.1" data-path="how-do-you-perform-a-genome-wide-snp-scan-to-generate-gwas-results.html"><a href="how-do-you-perform-a-genome-wide-snp-scan-to-generate-gwas-results.html#explanation-11"><i class="fa fa-check"></i><b>12.1</b> Explanation</a></li>
<li class="chapter" data-level="12.2" data-path="how-do-you-perform-a-genome-wide-snp-scan-to-generate-gwas-results.html"><a href="how-do-you-perform-a-genome-wide-snp-scan-to-generate-gwas-results.html#r-code-9"><i class="fa fa-check"></i><b>12.2</b> R Code</a></li>
</ul></li>
<li class="chapter" data-level="13" data-path="how-do-you-create-a-manhattan-plot-from-gwas-results-using-the-ggplot2-package.html"><a href="how-do-you-create-a-manhattan-plot-from-gwas-results-using-the-ggplot2-package.html"><i class="fa fa-check"></i><b>13</b> How do you create a Manhattan plot from GWAS results using the <code>ggplot2</code> package?</a>
<ul>
<li class="chapter" data-level="13.1" data-path="how-do-you-create-a-manhattan-plot-from-gwas-results-using-the-ggplot2-package.html"><a href="how-do-you-create-a-manhattan-plot-from-gwas-results-using-the-ggplot2-package.html#explanation-12"><i class="fa fa-check"></i><b>13.1</b> Explanation</a></li>
<li class="chapter" data-level="13.2" data-path="how-do-you-create-a-manhattan-plot-from-gwas-results-using-the-ggplot2-package.html"><a href="how-do-you-create-a-manhattan-plot-from-gwas-results-using-the-ggplot2-package.html#r-code-10"><i class="fa fa-check"></i><b>13.2</b> R Code</a></li>
</ul></li>
<li class="chapter" data-level="14" data-path="how-do-you-create-a-manhattan-plot-from-gwas-results-using-the-qqman-package.html"><a href="how-do-you-create-a-manhattan-plot-from-gwas-results-using-the-qqman-package.html"><i class="fa fa-check"></i><b>14</b> How do you create a Manhattan plot from GWAS results using the <code>qqman</code> package?</a>
<ul>
<li class="chapter" data-level="14.1" data-path="how-do-you-create-a-manhattan-plot-from-gwas-results-using-the-qqman-package.html"><a href="how-do-you-create-a-manhattan-plot-from-gwas-results-using-the-qqman-package.html#explanation-13"><i class="fa fa-check"></i><b>14.1</b> Explanation</a></li>
<li class="chapter" data-level="14.2" data-path="how-do-you-create-a-manhattan-plot-from-gwas-results-using-the-qqman-package.html"><a href="how-do-you-create-a-manhattan-plot-from-gwas-results-using-the-qqman-package.html#r-code-11"><i class="fa fa-check"></i><b>14.2</b> R Code</a></li>
</ul></li>
<li class="chapter" data-level="15" data-path="how-do-you-create-a-qq-plot-from-gwas-results-using-qqman-and-ggplot2.html"><a href="how-do-you-create-a-qq-plot-from-gwas-results-using-qqman-and-ggplot2.html"><i class="fa fa-check"></i><b>15</b> How do you create a QQ plot from GWAS results using <code>qqman</code> and <code>ggplot2</code>?</a>
<ul>
<li class="chapter" data-level="15.1" data-path="how-do-you-create-a-qq-plot-from-gwas-results-using-qqman-and-ggplot2.html"><a href="how-do-you-create-a-qq-plot-from-gwas-results-using-qqman-and-ggplot2.html#explanation-14"><i class="fa fa-check"></i><b>15.1</b> Explanation</a></li>
<li class="chapter" data-level="15.2" data-path="how-do-you-create-a-qq-plot-from-gwas-results-using-qqman-and-ggplot2.html"><a href="how-do-you-create-a-qq-plot-from-gwas-results-using-qqman-and-ggplot2.html#a.-using-the-qqman-package"><i class="fa fa-check"></i><b>15.2</b> A. Using the <code>qqman</code> package</a></li>
<li class="chapter" data-level="15.3" data-path="how-do-you-create-a-qq-plot-from-gwas-results-using-qqman-and-ggplot2.html"><a href="how-do-you-create-a-qq-plot-from-gwas-results-using-qqman-and-ggplot2.html#b.-using-ggplot2-for-more-control"><i class="fa fa-check"></i><b>15.3</b> B. Using <code>ggplot2</code> for more control</a></li>
</ul></li>
<li class="chapter" data-level="16" data-path="how-do-you-apply-multiple-testing-correction-to-gwas-results.html"><a href="how-do-you-apply-multiple-testing-correction-to-gwas-results.html"><i class="fa fa-check"></i><b>16</b> How do you apply multiple testing correction to GWAS results?</a>
<ul>
<li class="chapter" data-level="16.1" data-path="how-do-you-apply-multiple-testing-correction-to-gwas-results.html"><a href="how-do-you-apply-multiple-testing-correction-to-gwas-results.html#explanation-15"><i class="fa fa-check"></i><b>16.1</b> Explanation</a></li>
<li class="chapter" data-level="16.2" data-path="how-do-you-apply-multiple-testing-correction-to-gwas-results.html"><a href="how-do-you-apply-multiple-testing-correction-to-gwas-results.html#r-code-12"><i class="fa fa-check"></i><b>16.2</b> R Code</a></li>
<li class="chapter" data-level="16.3" data-path="how-do-you-apply-multiple-testing-correction-to-gwas-results.html"><a href="how-do-you-apply-multiple-testing-correction-to-gwas-results.html#interpretation"><i class="fa fa-check"></i><b>16.3</b> Interpretation</a>
<ul>
<li class="chapter" data-level="16.3.1" data-path="how-do-you-apply-multiple-testing-correction-to-gwas-results.html"><a href="how-do-you-apply-multiple-testing-correction-to-gwas-results.html#summary-table"><i class="fa fa-check"></i><b>16.3.1</b> Summary Table</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="17" data-path="how-do-you-create-a-volcano-plot-from-gwas-results-using-ggplot2.html"><a href="how-do-you-create-a-volcano-plot-from-gwas-results-using-ggplot2.html"><i class="fa fa-check"></i><b>17</b> How do you create a volcano plot from GWAS results using <code>ggplot2</code>?</a>
<ul>
<li class="chapter" data-level="17.1" data-path="how-do-you-create-a-volcano-plot-from-gwas-results-using-ggplot2.html"><a href="how-do-you-create-a-volcano-plot-from-gwas-results-using-ggplot2.html#explanation-16"><i class="fa fa-check"></i><b>17.1</b> Explanation</a></li>
<li class="chapter" data-level="17.2" data-path="how-do-you-create-a-volcano-plot-from-gwas-results-using-ggplot2.html"><a href="how-do-you-create-a-volcano-plot-from-gwas-results-using-ggplot2.html#r-code-13"><i class="fa fa-check"></i><b>17.2</b> R Code</a></li>
</ul></li>
<li class="chapter" data-level="18" data-path="how-do-you-identify-genome-wide-significant-snp-hits-and-save-them-for-downstream-analysis.html"><a href="how-do-you-identify-genome-wide-significant-snp-hits-and-save-them-for-downstream-analysis.html"><i class="fa fa-check"></i><b>18</b> How do you identify genome-wide significant SNP hits and save them for downstream analysis?</a>
<ul>
<li class="chapter" data-level="18.1" data-path="how-do-you-identify-genome-wide-significant-snp-hits-and-save-them-for-downstream-analysis.html"><a href="how-do-you-identify-genome-wide-significant-snp-hits-and-save-them-for-downstream-analysis.html#explanation-17"><i class="fa fa-check"></i><b>18.1</b> Explanation</a></li>
<li class="chapter" data-level="18.2" data-path="how-do-you-identify-genome-wide-significant-snp-hits-and-save-them-for-downstream-analysis.html"><a href="how-do-you-identify-genome-wide-significant-snp-hits-and-save-them-for-downstream-analysis.html#r-code-14"><i class="fa fa-check"></i><b>18.2</b> R Code</a></li>
</ul></li>
<li class="divider"></li>
<li><a href="https://complexdatainsights.com/product">Explore More Guides</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Learning GWAS One Step at a Time with R</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="how-do-you-tidy-the-genotype-matrix-from-a-.ped-file-in-r" class="section level1 hasAnchor" number="5">
<h1><span class="header-section-number">Q&A 5</span> How do you tidy the genotype matrix from a <code>.ped</code> file in R?<a href="how-do-you-tidy-the-genotype-matrix-from-a-.ped-file-in-r.html#how-do-you-tidy-the-genotype-matrix-from-a-.ped-file-in-r" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<div id="explanation-4" class="section level2 hasAnchor" number="5.1">
<h2><span class="header-section-number">5.1</span> Explanation<a href="how-do-you-tidy-the-genotype-matrix-from-a-.ped-file-in-r.html#explanation-4" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>PLINK <code>.ped</code> files store genotype data in a wide format, where each SNP is represented by <strong>two columns per individual</strong> (one for each allele). This structure is inefficient for downstream analysis, so we convert it to a <strong>tidy format</strong> where:</p>
<ul>
<li>Each row represents one sample</li>
<li>Each column represents one SNP</li>
<li>Alleles are combined (e.g., <code>"A A"</code>, <code>"A G"</code>, <code>"G G"</code>)</li>
</ul>
<p>We also use SNP metadata from the <code>.map</code> file ‚Äî now consistently referred to as <code>snp_info</code> ‚Äî to label the columns.</p>
</div>
<div id="r-code-3" class="section level2 hasAnchor" number="5.2">
<h2><span class="header-section-number">5.2</span> R Code<a href="how-do-you-tidy-the-genotype-matrix-from-a-.ped-file-in-r.html#r-code-3" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div class="sourceCode" id="cb27"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb27-1"><a href="how-do-you-tidy-the-genotype-matrix-from-a-.ped-file-in-r.html#cb27-1" tabindex="-1"></a><span class="co"># Load necessary library</span></span>
<span id="cb27-2"><a href="how-do-you-tidy-the-genotype-matrix-from-a-.ped-file-in-r.html#cb27-2" tabindex="-1"></a><span class="fu">library</span>(tidyverse)</span>
<span id="cb27-3"><a href="how-do-you-tidy-the-genotype-matrix-from-a-.ped-file-in-r.html#cb27-3" tabindex="-1"></a></span>
<span id="cb27-4"><a href="how-do-you-tidy-the-genotype-matrix-from-a-.ped-file-in-r.html#cb27-4" tabindex="-1"></a><span class="co"># Step 1: Load genotype and SNP info (if not already in memory)</span></span>
<span id="cb27-5"><a href="how-do-you-tidy-the-genotype-matrix-from-a-.ped-file-in-r.html#cb27-5" tabindex="-1"></a>ped_data <span class="ot">&lt;-</span> <span class="fu">read_rds</span>(<span class="st">&quot;data/sativa413.rds&quot;</span>)</span>
<span id="cb27-6"><a href="how-do-you-tidy-the-genotype-matrix-from-a-.ped-file-in-r.html#cb27-6" tabindex="-1"></a>snp_info <span class="ot">&lt;-</span> <span class="fu">read_table</span>(<span class="st">&quot;data/sativa413.map&quot;</span>, </span>
<span id="cb27-7"><a href="how-do-you-tidy-the-genotype-matrix-from-a-.ped-file-in-r.html#cb27-7" tabindex="-1"></a>                       <span class="at">col_names =</span> <span class="fu">c</span>(<span class="st">&quot;chr&quot;</span>, <span class="st">&quot;snp_id&quot;</span>, <span class="st">&quot;gen_dist&quot;</span>, <span class="st">&quot;bp_pos&quot;</span>), </span>
<span id="cb27-8"><a href="how-do-you-tidy-the-genotype-matrix-from-a-.ped-file-in-r.html#cb27-8" tabindex="-1"></a>                       <span class="at">show_col_types =</span> <span class="cn">FALSE</span>)</span>
<span id="cb27-9"><a href="how-do-you-tidy-the-genotype-matrix-from-a-.ped-file-in-r.html#cb27-9" tabindex="-1"></a></span>
<span id="cb27-10"><a href="how-do-you-tidy-the-genotype-matrix-from-a-.ped-file-in-r.html#cb27-10" tabindex="-1"></a><span class="co"># Step 2: Separate sample IDs and genotype calls</span></span>
<span id="cb27-11"><a href="how-do-you-tidy-the-genotype-matrix-from-a-.ped-file-in-r.html#cb27-11" tabindex="-1"></a>sample_ids <span class="ot">&lt;-</span> ped_data[, <span class="dv">1</span><span class="sc">:</span><span class="dv">2</span>]              <span class="co"># FID and IID</span></span>
<span id="cb27-12"><a href="how-do-you-tidy-the-genotype-matrix-from-a-.ped-file-in-r.html#cb27-12" tabindex="-1"></a>genotype_matrix <span class="ot">&lt;-</span> ped_data[, <span class="sc">-</span>(<span class="dv">1</span><span class="sc">:</span><span class="dv">6</span>)]      <span class="co"># Alleles only</span></span>
<span id="cb27-13"><a href="how-do-you-tidy-the-genotype-matrix-from-a-.ped-file-in-r.html#cb27-13" tabindex="-1"></a></span>
<span id="cb27-14"><a href="how-do-you-tidy-the-genotype-matrix-from-a-.ped-file-in-r.html#cb27-14" tabindex="-1"></a><span class="co"># Step 3: Verify expected SNP count</span></span>
<span id="cb27-15"><a href="how-do-you-tidy-the-genotype-matrix-from-a-.ped-file-in-r.html#cb27-15" tabindex="-1"></a>n_snps <span class="ot">&lt;-</span> <span class="fu">ncol</span>(genotype_matrix) <span class="sc">/</span> <span class="dv">2</span></span>
<span id="cb27-16"><a href="how-do-you-tidy-the-genotype-matrix-from-a-.ped-file-in-r.html#cb27-16" tabindex="-1"></a><span class="fu">stopifnot</span>(n_snps <span class="sc">==</span> <span class="fu">nrow</span>(snp_info))</span>
<span id="cb27-17"><a href="how-do-you-tidy-the-genotype-matrix-from-a-.ped-file-in-r.html#cb27-17" tabindex="-1"></a></span>
<span id="cb27-18"><a href="how-do-you-tidy-the-genotype-matrix-from-a-.ped-file-in-r.html#cb27-18" tabindex="-1"></a><span class="co"># Step 4: Combine each pair of allele columns into genotype strings</span></span>
<span id="cb27-19"><a href="how-do-you-tidy-the-genotype-matrix-from-a-.ped-file-in-r.html#cb27-19" tabindex="-1"></a>genotype_calls <span class="ot">&lt;-</span> <span class="fu">map_dfc</span>(<span class="fu">seq</span>(<span class="dv">1</span>, <span class="fu">ncol</span>(genotype_matrix), <span class="at">by =</span> <span class="dv">2</span>), <span class="cf">function</span>(i) {</span>
<span id="cb27-20"><a href="how-do-you-tidy-the-genotype-matrix-from-a-.ped-file-in-r.html#cb27-20" tabindex="-1"></a>  <span class="fu">paste</span>(genotype_matrix[[i]], genotype_matrix[[i <span class="sc">+</span> <span class="dv">1</span>]])</span>
<span id="cb27-21"><a href="how-do-you-tidy-the-genotype-matrix-from-a-.ped-file-in-r.html#cb27-21" tabindex="-1"></a>})</span>
<span id="cb27-22"><a href="how-do-you-tidy-the-genotype-matrix-from-a-.ped-file-in-r.html#cb27-22" tabindex="-1"></a><span class="fu">names</span>(genotype_calls) <span class="ot">&lt;-</span> snp_info<span class="sc">$</span>snp_id  <span class="co"># Use SNP IDs as column names</span></span>
<span id="cb27-23"><a href="how-do-you-tidy-the-genotype-matrix-from-a-.ped-file-in-r.html#cb27-23" tabindex="-1"></a></span>
<span id="cb27-24"><a href="how-do-you-tidy-the-genotype-matrix-from-a-.ped-file-in-r.html#cb27-24" tabindex="-1"></a><span class="co"># Step 5: Combine with sample IDs</span></span>
<span id="cb27-25"><a href="how-do-you-tidy-the-genotype-matrix-from-a-.ped-file-in-r.html#cb27-25" tabindex="-1"></a>genotype_tidy <span class="ot">&lt;-</span> <span class="fu">bind_cols</span>(sample_ids, genotype_calls)</span>
<span id="cb27-26"><a href="how-do-you-tidy-the-genotype-matrix-from-a-.ped-file-in-r.html#cb27-26" tabindex="-1"></a></span>
<span id="cb27-27"><a href="how-do-you-tidy-the-genotype-matrix-from-a-.ped-file-in-r.html#cb27-27" tabindex="-1"></a><span class="co"># Step 6: Preview output</span></span>
<span id="cb27-28"><a href="how-do-you-tidy-the-genotype-matrix-from-a-.ped-file-in-r.html#cb27-28" tabindex="-1"></a><span class="fu">head</span>(genotype_tidy[<span class="dv">1</span><span class="sc">:</span><span class="dv">10</span>, <span class="dv">1</span><span class="sc">:</span><span class="dv">5</span>])  <span class="co"># Show FID, IID, and first 8 SNPs</span></span></code></pre></div>
<pre><code># A tibble: 6 √ó 5
  X1            X2 id1000001 id1000003 id1000005
  &lt;chr&gt;      &lt;dbl&gt; &lt;chr&gt;     &lt;chr&gt;     &lt;chr&gt;    
1 081215-A05     1 T T       T T       C C      
2 081215-A06     3 C C       C C       C C      
3 081215-A07     4 C C       C C       C C      
4 081215-A08     5 C C       C C       T T      
5 090414-A09     6 C C       C C       C C      
6 090414-A10     7 T T       T T       C C      </code></pre>
<div class="sourceCode" id="cb29"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb29-1"><a href="how-do-you-tidy-the-genotype-matrix-from-a-.ped-file-in-r.html#cb29-1" tabindex="-1"></a><span class="fu">glimpse</span>(genotype_tidy[<span class="dv">1</span><span class="sc">:</span><span class="dv">5</span>, <span class="dv">1</span><span class="sc">:</span><span class="dv">10</span>])  <span class="co"># Show FID, IID, and first 8 SNPs</span></span></code></pre></div>
<pre><code>Rows: 5
Columns: 10
$ X1        &lt;chr&gt; &quot;081215-A05&quot;, &quot;081215-A06&quot;, &quot;081215-A07&quot;, &quot;081215-A08&quot;, &quot;090‚Ä¶
$ X2        &lt;dbl&gt; 1, 3, 4, 5, 6
$ id1000001 &lt;chr&gt; &quot;T T&quot;, &quot;C C&quot;, &quot;C C&quot;, &quot;C C&quot;, &quot;C C&quot;
$ id1000003 &lt;chr&gt; &quot;T T&quot;, &quot;C C&quot;, &quot;C C&quot;, &quot;C C&quot;, &quot;C C&quot;
$ id1000005 &lt;chr&gt; &quot;C C&quot;, &quot;C C&quot;, &quot;C C&quot;, &quot;T T&quot;, &quot;C C&quot;
$ id1000007 &lt;chr&gt; &quot;G G&quot;, &quot;A A&quot;, &quot;A A&quot;, &quot;G G&quot;, &quot;A A&quot;
$ id1000008 &lt;chr&gt; &quot;T T&quot;, &quot;G G&quot;, &quot;G G&quot;, &quot;G G&quot;, &quot;G G&quot;
$ id1000011 &lt;chr&gt; &quot;A A&quot;, &quot;0 0&quot;, &quot;G G&quot;, &quot;A A&quot;, &quot;G G&quot;
$ id1000013 &lt;chr&gt; &quot;C C&quot;, &quot;C C&quot;, &quot;C C&quot;, &quot;T T&quot;, &quot;C C&quot;
$ id1000015 &lt;chr&gt; &quot;T T&quot;, &quot;G G&quot;, &quot;G G&quot;, &quot;G G&quot;, &quot;G G&quot;</code></pre>
<blockquote>
<p>‚úÖ <strong>Takeaway:</strong> Tidying <code>.ped</code> genotype data into a clean sample-by-SNP format makes it much easier to analyze, visualize, or convert to numeric dosages. Use <code>snp_info</code> consistently as the SNP metadata reference to avoid conflicts.</p>
</blockquote>
</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="how-do-you-inspect-the-structure-and-contents-of-gwas-input-files-in-r.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="how-do-you-recode-allele-strings-into-numeric-count-format-for-gwas.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
  "sharing": {
    "github": false,
    "facebook": true,
    "twitter": true,
    "linkedin": false,
    "weibo": false,
    "instapaper": false,
    "vk": false,
    "whatsapp": false,
    "all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
  },
  "fontsettings": {
    "theme": "white",
    "family": "sans",
    "size": 2
  },
  "edit": {
    "link": "https://github.com/datainsights/cdi-gwas/edit/main/04-1-gwas.Rmd",
    "text": "Edit"
  },
  "history": {
    "link": null,
    "text": null
  },
  "view": {
    "link": null,
    "text": null
  },
  "download": ["gwas.pdf", "gwas.epub"],
  "search": {
    "engine": "fuse",
    "options": null
  },
  "toc": {
    "collapse": "subsection"
  }
});
});
</script>

</body>

</html>
