<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Q&A 6 How do you assign sample and SNP identifiers and align genotype data with phenotype records? | rnaseq.knit</title>
  <meta name="description" content="Genomics for Population Health" />
  <meta name="generator" content="bookdown 0.42 and GitBook 2.6.7" />

  <meta property="og:title" content="Q&A 6 How do you assign sample and SNP identifiers and align genotype data with phenotype records? | rnaseq.knit" />
  <meta property="og:type" content="book" />
  
  
  <meta name="github-repo" content="https://github.com/datainsights/cdi-gwas/" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Q&A 6 How do you assign sample and SNP identifiers and align genotype data with phenotype records? | rnaseq.knit" />
  
  
  




  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  <link rel="shortcut icon" href="favicon.ico" type="image/x-icon" />
<link rel="prev" href="how-do-you-recode-and-reshape-genotype-data-for-gwas-analysis.html"/>

<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { display: inline-block; text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
  
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="https://datainsights.github.io/cdi-gwas">Genome Wide Studies</a></li>

<li class="divider"></li>
<li><a href="index.html#section" id="toc-section"></a></li>
<li class="chapter" data-level="" data-path="welcome-to-cdi-gwas-for-population-health.html"><a href="welcome-to-cdi-gwas-for-population-health.html"><i class="fa fa-check"></i>Welcome to CDI ‚Äì GWAS for Population Health</a>
<ul>
<li class="chapter" data-level="" data-path="welcome-to-cdi-gwas-for-population-health.html"><a href="welcome-to-cdi-gwas-for-population-health.html#the-cdi-learning-path"><i class="fa fa-check"></i>üåê The CDI Learning Path</a></li>
</ul></li>
<li class="part"><span><b>I DATA EXPLORATION</b></span></li>
<li class="chapter" data-level="1" data-path="how-do-you-create-a-gwas-project-directory-ready-for-analysis.html"><a href="how-do-you-create-a-gwas-project-directory-ready-for-analysis.html"><i class="fa fa-check"></i><b>1</b> How do you create a GWAS project directory ready for analysis?</a>
<ul>
<li class="chapter" data-level="1.1" data-path="how-do-you-create-a-gwas-project-directory-ready-for-analysis.html"><a href="how-do-you-create-a-gwas-project-directory-ready-for-analysis.html#explanation"><i class="fa fa-check"></i><b>1.1</b> Explanation</a></li>
<li class="chapter" data-level="1.2" data-path="how-do-you-create-a-gwas-project-directory-ready-for-analysis.html"><a href="how-do-you-create-a-gwas-project-directory-ready-for-analysis.html#bash-terminal"><i class="fa fa-check"></i><b>1.2</b> Bash (Terminal)</a></li>
<li class="chapter" data-level="1.3" data-path="how-do-you-create-a-gwas-project-directory-ready-for-analysis.html"><a href="how-do-you-create-a-gwas-project-directory-ready-for-analysis.html#python-code"><i class="fa fa-check"></i><b>1.3</b> Python Code</a></li>
<li class="chapter" data-level="1.4" data-path="how-do-you-create-a-gwas-project-directory-ready-for-analysis.html"><a href="how-do-you-create-a-gwas-project-directory-ready-for-analysis.html#r-code"><i class="fa fa-check"></i><b>1.4</b> R Code</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="how-do-you-prepare-a-public-gwas-dataset-for-r-based-analysis.html"><a href="how-do-you-prepare-a-public-gwas-dataset-for-r-based-analysis.html"><i class="fa fa-check"></i><b>2</b> How do you prepare a public GWAS dataset for R-based analysis?</a>
<ul>
<li class="chapter" data-level="2.1" data-path="how-do-you-prepare-a-public-gwas-dataset-for-r-based-analysis.html"><a href="how-do-you-prepare-a-public-gwas-dataset-for-r-based-analysis.html#explanation-1"><i class="fa fa-check"></i><b>2.1</b> Explanation</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="how-do-you-efficiently-load-and-store-gwas-data-files-in-r.html"><a href="how-do-you-efficiently-load-and-store-gwas-data-files-in-r.html"><i class="fa fa-check"></i><b>3</b> How do you efficiently load and store GWAS data files in R?</a>
<ul>
<li class="chapter" data-level="3.1" data-path="how-do-you-efficiently-load-and-store-gwas-data-files-in-r.html"><a href="how-do-you-efficiently-load-and-store-gwas-data-files-in-r.html#explanation-2"><i class="fa fa-check"></i><b>3.1</b> Explanation</a></li>
<li class="chapter" data-level="3.2" data-path="how-do-you-efficiently-load-and-store-gwas-data-files-in-r.html"><a href="how-do-you-efficiently-load-and-store-gwas-data-files-in-r.html#r-code-1"><i class="fa fa-check"></i><b>3.2</b> R Code</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="how-do-you-inspect-the-structure-and-contents-of-gwas-input-files-in-r.html"><a href="how-do-you-inspect-the-structure-and-contents-of-gwas-input-files-in-r.html"><i class="fa fa-check"></i><b>4</b> How do you inspect the structure and contents of GWAS input files in R?</a>
<ul>
<li class="chapter" data-level="4.1" data-path="how-do-you-inspect-the-structure-and-contents-of-gwas-input-files-in-r.html"><a href="how-do-you-inspect-the-structure-and-contents-of-gwas-input-files-in-r.html#explanation-3"><i class="fa fa-check"></i><b>4.1</b> Explanation</a></li>
<li class="chapter" data-level="4.2" data-path="how-do-you-inspect-the-structure-and-contents-of-gwas-input-files-in-r.html"><a href="how-do-you-inspect-the-structure-and-contents-of-gwas-input-files-in-r.html#r-code-2"><i class="fa fa-check"></i><b>4.2</b> R Code</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="how-do-you-recode-and-reshape-genotype-data-for-gwas-analysis.html"><a href="how-do-you-recode-and-reshape-genotype-data-for-gwas-analysis.html"><i class="fa fa-check"></i><b>5</b> How do you recode and reshape genotype data for GWAS analysis?</a>
<ul>
<li class="chapter" data-level="5.1" data-path="how-do-you-recode-and-reshape-genotype-data-for-gwas-analysis.html"><a href="how-do-you-recode-and-reshape-genotype-data-for-gwas-analysis.html#explanation-4"><i class="fa fa-check"></i><b>5.1</b> Explanation</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="how-do-you-assign-sample-and-snp-identifiers-and-align-genotype-data-with-phenotype-records.html"><a href="how-do-you-assign-sample-and-snp-identifiers-and-align-genotype-data-with-phenotype-records.html"><i class="fa fa-check"></i><b>6</b> How do you assign sample and SNP identifiers and align genotype data with phenotype records?</a>
<ul>
<li class="chapter" data-level="6.1" data-path="how-do-you-assign-sample-and-snp-identifiers-and-align-genotype-data-with-phenotype-records.html"><a href="how-do-you-assign-sample-and-snp-identifiers-and-align-genotype-data-with-phenotype-records.html#explanation-5"><i class="fa fa-check"></i><b>6.1</b> Explanation</a></li>
<li class="chapter" data-level="6.2" data-path="how-do-you-assign-sample-and-snp-identifiers-and-align-genotype-data-with-phenotype-records.html"><a href="how-do-you-assign-sample-and-snp-identifiers-and-align-genotype-data-with-phenotype-records.html#r-code-3"><i class="fa fa-check"></i><b>6.2</b> R Code</a></li>
</ul></li>
<li class="divider"></li>
<li><a href="https://complexdatainsights.com/product">Explore More Guides</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./"><div class="line-block">Genomics for Population Health</div></a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="how-do-you-assign-sample-and-snp-identifiers-and-align-genotype-data-with-phenotype-records" class="section level1 hasAnchor" number="6">
<h1><span class="header-section-number">Q&A 6</span> How do you assign sample and SNP identifiers and align genotype data with phenotype records?<a href="how-do-you-assign-sample-and-snp-identifiers-and-align-genotype-data-with-phenotype-records.html#how-do-you-assign-sample-and-snp-identifiers-and-align-genotype-data-with-phenotype-records" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<div id="explanation-5" class="section level2 hasAnchor" number="6.1">
<h2><span class="header-section-number">6.1</span> Explanation<a href="how-do-you-assign-sample-and-snp-identifiers-and-align-genotype-data-with-phenotype-records.html#explanation-5" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>After converting genotype data to a matrix, the next step is to label rows and columns using the original <code>.fam</code> and <code>.map</code> files:</p>
<ul>
<li><code>.fam</code> provides sample identifiers (usually the second column = <code>IID</code>)</li>
<li><code>.map</code> provides SNP identifiers (usually the second column = <code>SNP ID</code>)</li>
</ul>
<p>We‚Äôll load those files, assign them as <code>rownames()</code> and <code>colnames()</code>, and then verify that the sample IDs match the phenotype data ‚Äî ensuring both data types are aligned for analysis.</p>
</div>
<div id="r-code-3" class="section level2 hasAnchor" number="6.2">
<h2><span class="header-section-number">6.2</span> R Code<a href="how-do-you-assign-sample-and-snp-identifiers-and-align-genotype-data-with-phenotype-records.html#r-code-3" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div class="sourceCode" id="cb22"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb22-1"><a href="how-do-you-assign-sample-and-snp-identifiers-and-align-genotype-data-with-phenotype-records.html#cb22-1" tabindex="-1"></a><span class="fu">library</span>(tidyverse)</span>
<span id="cb22-2"><a href="how-do-you-assign-sample-and-snp-identifiers-and-align-genotype-data-with-phenotype-records.html#cb22-2" tabindex="-1"></a><span class="fu">library</span>(rrBLUP)</span>
<span id="cb22-3"><a href="how-do-you-assign-sample-and-snp-identifiers-and-align-genotype-data-with-phenotype-records.html#cb22-3" tabindex="-1"></a><span class="fu">library</span>(BGLR)</span>
<span id="cb22-4"><a href="how-do-you-assign-sample-and-snp-identifiers-and-align-genotype-data-with-phenotype-records.html#cb22-4" tabindex="-1"></a><span class="fu">library</span>(DT)</span>
<span id="cb22-5"><a href="how-do-you-assign-sample-and-snp-identifiers-and-align-genotype-data-with-phenotype-records.html#cb22-5" tabindex="-1"></a><span class="fu">library</span>(SNPRelate)</span>
<span id="cb22-6"><a href="how-do-you-assign-sample-and-snp-identifiers-and-align-genotype-data-with-phenotype-records.html#cb22-6" tabindex="-1"></a><span class="fu">library</span>(dplyr)</span>
<span id="cb22-7"><a href="how-do-you-assign-sample-and-snp-identifiers-and-align-genotype-data-with-phenotype-records.html#cb22-7" tabindex="-1"></a><span class="fu">library</span>(qqman)</span>
<span id="cb22-8"><a href="how-do-you-assign-sample-and-snp-identifiers-and-align-genotype-data-with-phenotype-records.html#cb22-8" tabindex="-1"></a><span class="fu">library</span>(poolr)</span>
<span id="cb22-9"><a href="how-do-you-assign-sample-and-snp-identifiers-and-align-genotype-data-with-phenotype-records.html#cb22-9" tabindex="-1"></a></span>
<span id="cb22-10"><a href="how-do-you-assign-sample-and-snp-identifiers-and-align-genotype-data-with-phenotype-records.html#cb22-10" tabindex="-1"></a><span class="co"># üß¨ Load genotype matrix</span></span>
<span id="cb22-11"><a href="how-do-you-assign-sample-and-snp-identifiers-and-align-genotype-data-with-phenotype-records.html#cb22-11" tabindex="-1"></a>Geno <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(<span class="st">&quot;RDataRDS/geno_matrix.rds&quot;</span>)</span>
<span id="cb22-12"><a href="how-do-you-assign-sample-and-snp-identifiers-and-align-genotype-data-with-phenotype-records.html#cb22-12" tabindex="-1"></a></span>
<span id="cb22-13"><a href="how-do-you-assign-sample-and-snp-identifiers-and-align-genotype-data-with-phenotype-records.html#cb22-13" tabindex="-1"></a><span class="co"># üìë Load sample and SNP info</span></span>
<span id="cb22-14"><a href="how-do-you-assign-sample-and-snp-identifiers-and-align-genotype-data-with-phenotype-records.html#cb22-14" tabindex="-1"></a>FAM <span class="ot">&lt;-</span> <span class="fu">read_table</span>(<span class="st">&quot;data/sativa413.fam&quot;</span>, <span class="at">col_names =</span> <span class="cn">FALSE</span>, <span class="at">show_col_types =</span> <span class="cn">FALSE</span>)</span>
<span id="cb22-15"><a href="how-do-you-assign-sample-and-snp-identifiers-and-align-genotype-data-with-phenotype-records.html#cb22-15" tabindex="-1"></a>MAP <span class="ot">&lt;-</span> <span class="fu">read_table</span>(<span class="st">&quot;data/sativa413.map&quot;</span>, <span class="at">col_names =</span> <span class="cn">FALSE</span>, <span class="at">show_col_types =</span> <span class="cn">FALSE</span>)</span>
<span id="cb22-16"><a href="how-do-you-assign-sample-and-snp-identifiers-and-align-genotype-data-with-phenotype-records.html#cb22-16" tabindex="-1"></a></span>
<span id="cb22-17"><a href="how-do-you-assign-sample-and-snp-identifiers-and-align-genotype-data-with-phenotype-records.html#cb22-17" tabindex="-1"></a><span class="co"># üè∑Ô∏è Assign row and column names</span></span>
<span id="cb22-18"><a href="how-do-you-assign-sample-and-snp-identifiers-and-align-genotype-data-with-phenotype-records.html#cb22-18" tabindex="-1"></a><span class="fu">rownames</span>(Geno) <span class="ot">&lt;-</span> FAM<span class="sc">$</span>X2        <span class="co"># Individual IDs</span></span>
<span id="cb22-19"><a href="how-do-you-assign-sample-and-snp-identifiers-and-align-genotype-data-with-phenotype-records.html#cb22-19" tabindex="-1"></a><span class="fu">colnames</span>(Geno) <span class="ot">&lt;-</span> MAP<span class="sc">$</span>X2        <span class="co"># SNP IDs</span></span>
<span id="cb22-20"><a href="how-do-you-assign-sample-and-snp-identifiers-and-align-genotype-data-with-phenotype-records.html#cb22-20" tabindex="-1"></a></span>
<span id="cb22-21"><a href="how-do-you-assign-sample-and-snp-identifiers-and-align-genotype-data-with-phenotype-records.html#cb22-21" tabindex="-1"></a><span class="co"># üìÑ Load phenotype data</span></span>
<span id="cb22-22"><a href="how-do-you-assign-sample-and-snp-identifiers-and-align-genotype-data-with-phenotype-records.html#cb22-22" tabindex="-1"></a>phenotypes <span class="ot">&lt;-</span> <span class="fu">read_delim</span>(<span class="st">&quot;data/sativa_phenotypes.tsv&quot;</span>, <span class="at">show_col_types =</span> <span class="cn">FALSE</span>)</span>
<span id="cb22-23"><a href="how-do-you-assign-sample-and-snp-identifiers-and-align-genotype-data-with-phenotype-records.html#cb22-23" tabindex="-1"></a></span>
<span id="cb22-24"><a href="how-do-you-assign-sample-and-snp-identifiers-and-align-genotype-data-with-phenotype-records.html#cb22-24" tabindex="-1"></a><span class="co"># ‚úÖ Check alignment between genotype matrix and phenotype table</span></span>
<span id="cb22-25"><a href="how-do-you-assign-sample-and-snp-identifiers-and-align-genotype-data-with-phenotype-records.html#cb22-25" tabindex="-1"></a><span class="fu">table</span>(<span class="fu">rownames</span>(Geno) <span class="sc">==</span> phenotypes<span class="sc">$</span>NSFTVID)</span></code></pre></div>
<blockquote>
<p>‚úÖ <strong>Takeaway:</strong> Assigning identifiers from <code>.fam</code> and <code>.map</code> ensures interpretability of your genotype matrix. Verifying alignment with the phenotype table prevents mismatches and guarantees that downstream analyses are valid and reproducible.</p>
</blockquote>

</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="how-do-you-recode-and-reshape-genotype-data-for-gwas-analysis.html" class="navigation navigation-prev navigation-unique" aria-label="Previous page"><i class="fa fa-angle-left"></i></a>

    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
  "sharing": {
    "github": false,
    "facebook": true,
    "twitter": true,
    "linkedin": false,
    "weibo": false,
    "instapaper": false,
    "vk": false,
    "whatsapp": false,
    "all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
  },
  "fontsettings": {
    "theme": "white",
    "family": "sans",
    "size": 2
  },
  "edit": {
    "link": "https://github.com/datainsights/cdi-gwas/edit/main/03-1-gwas.Rmd",
    "text": "Edit"
  },
  "history": {
    "link": null,
    "text": null
  },
  "view": {
    "link": null,
    "text": null
  },
  "download": ["rnaseq.pdf", "rnaseq.epub"],
  "search": {
    "engine": "fuse",
    "options": null
  },
  "toc": {
    "collapse": "subsection"
  }
});
});
</script>

</body>

</html>
