<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Q&A 11 How do you include PCA covariates in a GWAS model? | gwas.knit</title>
  <meta name="description" content="Genomics for Population Health" />
  <meta name="generator" content="bookdown 0.42 and GitBook 2.6.7" />

  <meta property="og:title" content="Q&A 11 How do you include PCA covariates in a GWAS model? | gwas.knit" />
  <meta property="og:type" content="book" />
  
  
  <meta name="github-repo" content="https://github.com/datainsights/cdi-gwas/" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Q&A 11 How do you include PCA covariates in a GWAS model? | gwas.knit" />
  
  
  




  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  <link rel="shortcut icon" href="favicon.ico" type="image/x-icon" />
<link rel="prev" href="how-do-you-perform-pca-on-genotype-data-to-assess-population-structure.html"/>
<link rel="next" href="how-do-you-interpret-gwas-model-results-with-pca-covariates.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { display: inline-block; text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
  
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="https://datainsights.github.io/cdi-gwas">Genome Wide Studies</a></li>

<li class="divider"></li>
<li><a href="index.html#section" id="toc-section"></a></li>
<li class="chapter" data-level="" data-path="welcome-to-cdi-gwas-for-population-health.html"><a href="welcome-to-cdi-gwas-for-population-health.html"><i class="fa fa-check"></i>Welcome to CDI ‚Äì GWAS for Population Health</a>
<ul>
<li class="chapter" data-level="" data-path="welcome-to-cdi-gwas-for-population-health.html"><a href="welcome-to-cdi-gwas-for-population-health.html#the-cdi-learning-path"><i class="fa fa-check"></i>üåê The CDI Learning Path</a></li>
</ul></li>
<li class="part"><span><b>I DATA EXPLORATION</b></span></li>
<li class="chapter" data-level="1" data-path="how-do-you-create-a-gwas-project-directory-ready-for-analysis.html"><a href="how-do-you-create-a-gwas-project-directory-ready-for-analysis.html"><i class="fa fa-check"></i><b>1</b> How do you create a GWAS project directory ready for analysis?</a>
<ul>
<li class="chapter" data-level="1.1" data-path="how-do-you-create-a-gwas-project-directory-ready-for-analysis.html"><a href="how-do-you-create-a-gwas-project-directory-ready-for-analysis.html#explanation"><i class="fa fa-check"></i><b>1.1</b> Explanation</a></li>
<li class="chapter" data-level="1.2" data-path="how-do-you-create-a-gwas-project-directory-ready-for-analysis.html"><a href="how-do-you-create-a-gwas-project-directory-ready-for-analysis.html#bash-terminal"><i class="fa fa-check"></i><b>1.2</b> Bash (Terminal)</a></li>
<li class="chapter" data-level="1.3" data-path="how-do-you-create-a-gwas-project-directory-ready-for-analysis.html"><a href="how-do-you-create-a-gwas-project-directory-ready-for-analysis.html#python-code"><i class="fa fa-check"></i><b>1.3</b> Python Code</a></li>
<li class="chapter" data-level="1.4" data-path="how-do-you-create-a-gwas-project-directory-ready-for-analysis.html"><a href="how-do-you-create-a-gwas-project-directory-ready-for-analysis.html#r-code"><i class="fa fa-check"></i><b>1.4</b> R Code</a></li>
<li class="chapter" data-level="1.5" data-path="how-do-you-create-a-gwas-project-directory-ready-for-analysis.html"><a href="how-do-you-create-a-gwas-project-directory-ready-for-analysis.html#import-libraries"><i class="fa fa-check"></i><b>1.5</b> Import libraries</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="how-do-you-prepare-a-public-gwas-dataset-for-r-based-analysis.html"><a href="how-do-you-prepare-a-public-gwas-dataset-for-r-based-analysis.html"><i class="fa fa-check"></i><b>2</b> How do you prepare a public GWAS dataset for R-based analysis?</a>
<ul>
<li class="chapter" data-level="2.1" data-path="how-do-you-prepare-a-public-gwas-dataset-for-r-based-analysis.html"><a href="how-do-you-prepare-a-public-gwas-dataset-for-r-based-analysis.html#explanation-1"><i class="fa fa-check"></i><b>2.1</b> Explanation</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="how-do-you-prepare-a-public-gwas-dataset-for-r-based-analysis-1.html"><a href="how-do-you-prepare-a-public-gwas-dataset-for-r-based-analysis-1.html"><i class="fa fa-check"></i><b>3</b> How do you prepare a public GWAS dataset for R-based analysis?</a>
<ul>
<li class="chapter" data-level="3.1" data-path="how-do-you-prepare-a-public-gwas-dataset-for-r-based-analysis-1.html"><a href="how-do-you-prepare-a-public-gwas-dataset-for-r-based-analysis-1.html#explanation-2"><i class="fa fa-check"></i><b>3.1</b> Explanation</a></li>
<li class="chapter" data-level="3.2" data-path="how-do-you-prepare-a-public-gwas-dataset-for-r-based-analysis-1.html"><a href="how-do-you-prepare-a-public-gwas-dataset-for-r-based-analysis-1.html#bash-script"><i class="fa fa-check"></i><b>3.2</b> Bash Script</a></li>
<li class="chapter" data-level="3.3" data-path="how-do-you-prepare-a-public-gwas-dataset-for-r-based-analysis-1.html"><a href="how-do-you-prepare-a-public-gwas-dataset-for-r-based-analysis-1.html#file-structure"><i class="fa fa-check"></i><b>3.3</b> File Structure</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="how-do-you-efficiently-load-and-store-gwas-data-files-in-r.html"><a href="how-do-you-efficiently-load-and-store-gwas-data-files-in-r.html"><i class="fa fa-check"></i><b>4</b> How do you efficiently load and store GWAS data files in R?</a>
<ul>
<li class="chapter" data-level="4.1" data-path="how-do-you-efficiently-load-and-store-gwas-data-files-in-r.html"><a href="how-do-you-efficiently-load-and-store-gwas-data-files-in-r.html#explanation-3"><i class="fa fa-check"></i><b>4.1</b> Explanation</a></li>
<li class="chapter" data-level="4.2" data-path="how-do-you-efficiently-load-and-store-gwas-data-files-in-r.html"><a href="how-do-you-efficiently-load-and-store-gwas-data-files-in-r.html#r-code-1"><i class="fa fa-check"></i><b>4.2</b> R Code</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="how-do-you-inspect-the-structure-and-contents-of-gwas-input-files-in-r.html"><a href="how-do-you-inspect-the-structure-and-contents-of-gwas-input-files-in-r.html"><i class="fa fa-check"></i><b>5</b> How do you inspect the structure and contents of GWAS input files in R?</a>
<ul>
<li class="chapter" data-level="5.1" data-path="how-do-you-inspect-the-structure-and-contents-of-gwas-input-files-in-r.html"><a href="how-do-you-inspect-the-structure-and-contents-of-gwas-input-files-in-r.html#explanation-4"><i class="fa fa-check"></i><b>5.1</b> Explanation</a></li>
<li class="chapter" data-level="5.2" data-path="how-do-you-inspect-the-structure-and-contents-of-gwas-input-files-in-r.html"><a href="how-do-you-inspect-the-structure-and-contents-of-gwas-input-files-in-r.html#r-code-2"><i class="fa fa-check"></i><b>5.2</b> R Code</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="how-do-you-tidy-the-genotype-matrix-from-a-.ped-file-in-r.html"><a href="how-do-you-tidy-the-genotype-matrix-from-a-.ped-file-in-r.html"><i class="fa fa-check"></i><b>6</b> How do you tidy the genotype matrix from a <code>.ped</code> file in R?</a>
<ul>
<li class="chapter" data-level="6.1" data-path="how-do-you-tidy-the-genotype-matrix-from-a-.ped-file-in-r.html"><a href="how-do-you-tidy-the-genotype-matrix-from-a-.ped-file-in-r.html#explanation-5"><i class="fa fa-check"></i><b>6.1</b> Explanation</a></li>
<li class="chapter" data-level="6.2" data-path="how-do-you-tidy-the-genotype-matrix-from-a-.ped-file-in-r.html"><a href="how-do-you-tidy-the-genotype-matrix-from-a-.ped-file-in-r.html#r-code-3"><i class="fa fa-check"></i><b>6.2</b> R Code</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="how-do-you-recode-allele-strings-into-numeric-count-format-for-gwas.html"><a href="how-do-you-recode-allele-strings-into-numeric-count-format-for-gwas.html"><i class="fa fa-check"></i><b>7</b> How do you recode allele strings into numeric count format for GWAS?</a>
<ul>
<li class="chapter" data-level="7.1" data-path="how-do-you-recode-allele-strings-into-numeric-count-format-for-gwas.html"><a href="how-do-you-recode-allele-strings-into-numeric-count-format-for-gwas.html#explanation-6"><i class="fa fa-check"></i><b>7.1</b> Explanation</a></li>
<li class="chapter" data-level="7.2" data-path="how-do-you-recode-allele-strings-into-numeric-count-format-for-gwas.html"><a href="how-do-you-recode-allele-strings-into-numeric-count-format-for-gwas.html#r-code-4"><i class="fa fa-check"></i><b>7.2</b> R Code</a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="how-do-you-filter-snps-and-samples-based-on-missing-data-and-minor-allele-frequency-how-do-you-filter-genotype-data-for-missingness-and-minor-allele-frequency.html"><a href="how-do-you-filter-snps-and-samples-based-on-missing-data-and-minor-allele-frequency-how-do-you-filter-genotype-data-for-missingness-and-minor-allele-frequency.html"><i class="fa fa-check"></i><b>8</b> How do you filter SNPs and samples based on missing data and minor allele frequency?# How do you filter genotype data for missingness and minor allele frequency?</a>
<ul>
<li class="chapter" data-level="8.1" data-path="how-do-you-filter-snps-and-samples-based-on-missing-data-and-minor-allele-frequency-how-do-you-filter-genotype-data-for-missingness-and-minor-allele-frequency.html"><a href="how-do-you-filter-snps-and-samples-based-on-missing-data-and-minor-allele-frequency-how-do-you-filter-genotype-data-for-missingness-and-minor-allele-frequency.html#explanation-7"><i class="fa fa-check"></i><b>8.1</b> Explanation</a></li>
<li class="chapter" data-level="8.2" data-path="how-do-you-filter-snps-and-samples-based-on-missing-data-and-minor-allele-frequency-how-do-you-filter-genotype-data-for-missingness-and-minor-allele-frequency.html"><a href="how-do-you-filter-snps-and-samples-based-on-missing-data-and-minor-allele-frequency-how-do-you-filter-genotype-data-for-missingness-and-minor-allele-frequency.html#r-code-5"><i class="fa fa-check"></i><b>8.2</b> R Code</a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="how-do-you-impute-missing-genotype-values-before-gwas-analysis.html"><a href="how-do-you-impute-missing-genotype-values-before-gwas-analysis.html"><i class="fa fa-check"></i><b>9</b> How do you impute missing genotype values before GWAS analysis?</a>
<ul>
<li class="chapter" data-level="9.1" data-path="how-do-you-impute-missing-genotype-values-before-gwas-analysis.html"><a href="how-do-you-impute-missing-genotype-values-before-gwas-analysis.html#explanation-8"><i class="fa fa-check"></i><b>9.1</b> Explanation</a></li>
<li class="chapter" data-level="9.2" data-path="how-do-you-impute-missing-genotype-values-before-gwas-analysis.html"><a href="how-do-you-impute-missing-genotype-values-before-gwas-analysis.html#r-code-6"><i class="fa fa-check"></i><b>9.2</b> R Code</a></li>
</ul></li>
<li class="chapter" data-level="10" data-path="how-do-you-perform-pca-on-genotype-data-to-assess-population-structure.html"><a href="how-do-you-perform-pca-on-genotype-data-to-assess-population-structure.html"><i class="fa fa-check"></i><b>10</b> How do you perform PCA on genotype data to assess population structure?</a>
<ul>
<li class="chapter" data-level="10.1" data-path="how-do-you-perform-pca-on-genotype-data-to-assess-population-structure.html"><a href="how-do-you-perform-pca-on-genotype-data-to-assess-population-structure.html#explanation-9"><i class="fa fa-check"></i><b>10.1</b> Explanation</a></li>
<li class="chapter" data-level="10.2" data-path="how-do-you-perform-pca-on-genotype-data-to-assess-population-structure.html"><a href="how-do-you-perform-pca-on-genotype-data-to-assess-population-structure.html#r-code-7"><i class="fa fa-check"></i><b>10.2</b> R Code</a></li>
</ul></li>
<li class="chapter" data-level="11" data-path="how-do-you-include-pca-covariates-in-a-gwas-model.html"><a href="how-do-you-include-pca-covariates-in-a-gwas-model.html"><i class="fa fa-check"></i><b>11</b> How do you include PCA covariates in a GWAS model?</a>
<ul>
<li class="chapter" data-level="11.1" data-path="how-do-you-include-pca-covariates-in-a-gwas-model.html"><a href="how-do-you-include-pca-covariates-in-a-gwas-model.html#explanation-10"><i class="fa fa-check"></i><b>11.1</b> Explanation</a></li>
<li class="chapter" data-level="11.2" data-path="how-do-you-include-pca-covariates-in-a-gwas-model.html"><a href="how-do-you-include-pca-covariates-in-a-gwas-model.html#r-code-8"><i class="fa fa-check"></i><b>11.2</b> R Code</a></li>
</ul></li>
<li class="chapter" data-level="12" data-path="how-do-you-interpret-gwas-model-results-with-pca-covariates.html"><a href="how-do-you-interpret-gwas-model-results-with-pca-covariates.html"><i class="fa fa-check"></i><b>12</b> How do you interpret GWAS model results with PCA covariates?</a>
<ul>
<li class="chapter" data-level="12.1" data-path="how-do-you-interpret-gwas-model-results-with-pca-covariates.html"><a href="how-do-you-interpret-gwas-model-results-with-pca-covariates.html#explanation-11"><i class="fa fa-check"></i><b>12.1</b> Explanation</a></li>
<li class="chapter" data-level="12.2" data-path="how-do-you-interpret-gwas-model-results-with-pca-covariates.html"><a href="how-do-you-interpret-gwas-model-results-with-pca-covariates.html#r-model-output-summary"><i class="fa fa-check"></i><b>12.2</b> R Model Output Summary</a></li>
</ul></li>
<li class="divider"></li>
<li><a href="https://complexdatainsights.com/product">Explore More Guides</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./"><div class="line-block">Genomics for Population Health</div></a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="how-do-you-include-pca-covariates-in-a-gwas-model" class="section level1 hasAnchor" number="11">
<h1><span class="header-section-number">Q&A 11</span> How do you include PCA covariates in a GWAS model?<a href="how-do-you-include-pca-covariates-in-a-gwas-model.html#how-do-you-include-pca-covariates-in-a-gwas-model" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<div id="explanation-10" class="section level2 hasAnchor" number="11.1">
<h2><span class="header-section-number">11.1</span> Explanation<a href="how-do-you-include-pca-covariates-in-a-gwas-model.html#explanation-10" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>To reduce false positives caused by hidden genetic structure, it‚Äôs standard in GWAS to include the top principal components (PCs) as covariates in the model. These PCs come from PCA applied to the genotype matrix and must be matched back to each sample using the same identifiers (<code>FID</code>, <code>IID</code>).</p>
<p>This Q&amp;A walks through the process of combining phenotype, PCA, and genotype data, then fitting a single SNP-trait association model.</p>
</div>
<div id="r-code-8" class="section level2 hasAnchor" number="11.2">
<h2><span class="header-section-number">11.2</span> R Code<a href="how-do-you-include-pca-covariates-in-a-gwas-model.html#r-code-8" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div class="sourceCode" id="cb44"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb44-1"><a href="how-do-you-include-pca-covariates-in-a-gwas-model.html#cb44-1" tabindex="-1"></a><span class="co"># Load libraries</span></span>
<span id="cb44-2"><a href="how-do-you-include-pca-covariates-in-a-gwas-model.html#cb44-2" tabindex="-1"></a><span class="fu">library</span>(tidyverse)</span>
<span id="cb44-3"><a href="how-do-you-include-pca-covariates-in-a-gwas-model.html#cb44-3" tabindex="-1"></a></span>
<span id="cb44-4"><a href="how-do-you-include-pca-covariates-in-a-gwas-model.html#cb44-4" tabindex="-1"></a><span class="co"># Step 1: Load and align phenotype data with sample metadata (by FID)</span></span>
<span id="cb44-5"><a href="how-do-you-include-pca-covariates-in-a-gwas-model.html#cb44-5" tabindex="-1"></a>fam_data <span class="ot">&lt;-</span> <span class="fu">read_table</span>(<span class="st">&quot;data/sativas413.fam&quot;</span>, </span>
<span id="cb44-6"><a href="how-do-you-include-pca-covariates-in-a-gwas-model.html#cb44-6" tabindex="-1"></a>                       <span class="at">col_names =</span> <span class="fu">c</span>(<span class="st">&quot;FID&quot;</span>, <span class="st">&quot;IID&quot;</span>, <span class="st">&quot;PID&quot;</span>, <span class="st">&quot;MID&quot;</span>, <span class="st">&quot;sex&quot;</span>, <span class="st">&quot;phenotype&quot;</span>), </span>
<span id="cb44-7"><a href="how-do-you-include-pca-covariates-in-a-gwas-model.html#cb44-7" tabindex="-1"></a>                       <span class="at">show_col_types =</span> <span class="cn">FALSE</span>)</span>
<span id="cb44-8"><a href="how-do-you-include-pca-covariates-in-a-gwas-model.html#cb44-8" tabindex="-1"></a></span>
<span id="cb44-9"><a href="how-do-you-include-pca-covariates-in-a-gwas-model.html#cb44-9" tabindex="-1"></a>phenotype_data <span class="ot">&lt;-</span> <span class="fu">read_tsv</span>(<span class="st">&quot;data/sativa413_phenotypes.txt&quot;</span>, <span class="at">show_col_types =</span> <span class="cn">FALSE</span>) <span class="sc">%&gt;%</span></span>
<span id="cb44-10"><a href="how-do-you-include-pca-covariates-in-a-gwas-model.html#cb44-10" tabindex="-1"></a>  <span class="fu">rename</span>(<span class="at">FID =</span> <span class="dv">1</span>)  <span class="co"># Sample IDs in phenotype file match fam_data$FID</span></span>
<span id="cb44-11"><a href="how-do-you-include-pca-covariates-in-a-gwas-model.html#cb44-11" tabindex="-1"></a></span>
<span id="cb44-12"><a href="how-do-you-include-pca-covariates-in-a-gwas-model.html#cb44-12" tabindex="-1"></a>sample_metadata <span class="ot">&lt;-</span> <span class="fu">left_join</span>(fam_data, phenotype_data, <span class="at">by =</span> <span class="st">&quot;FID&quot;</span>)</span>
<span id="cb44-13"><a href="how-do-you-include-pca-covariates-in-a-gwas-model.html#cb44-13" tabindex="-1"></a></span>
<span id="cb44-14"><a href="how-do-you-include-pca-covariates-in-a-gwas-model.html#cb44-14" tabindex="-1"></a><span class="co"># Step 2: Rename PCA columns to include proper IDs</span></span>
<span id="cb44-15"><a href="how-do-you-include-pca-covariates-in-a-gwas-model.html#cb44-15" tabindex="-1"></a>pca_df <span class="ot">&lt;-</span> pca_df <span class="sc">%&gt;%</span></span>
<span id="cb44-16"><a href="how-do-you-include-pca-covariates-in-a-gwas-model.html#cb44-16" tabindex="-1"></a>  <span class="fu">rename</span>(<span class="at">FID =</span> <span class="dv">1</span>, <span class="at">IID =</span> <span class="dv">2</span>)</span>
<span id="cb44-17"><a href="how-do-you-include-pca-covariates-in-a-gwas-model.html#cb44-17" tabindex="-1"></a></span>
<span id="cb44-18"><a href="how-do-you-include-pca-covariates-in-a-gwas-model.html#cb44-18" tabindex="-1"></a><span class="co"># Step 3: Merge PCA data into sample metadata</span></span>
<span id="cb44-19"><a href="how-do-you-include-pca-covariates-in-a-gwas-model.html#cb44-19" tabindex="-1"></a>sample_data <span class="ot">&lt;-</span> <span class="fu">left_join</span>(sample_metadata, pca_df, <span class="at">by =</span> <span class="fu">c</span>(<span class="st">&quot;FID&quot;</span>, <span class="st">&quot;IID&quot;</span>))</span>
<span id="cb44-20"><a href="how-do-you-include-pca-covariates-in-a-gwas-model.html#cb44-20" tabindex="-1"></a></span>
<span id="cb44-21"><a href="how-do-you-include-pca-covariates-in-a-gwas-model.html#cb44-21" tabindex="-1"></a><span class="co"># Step 4: Standardize ID columns in genotype data</span></span>
<span id="cb44-22"><a href="how-do-you-include-pca-covariates-in-a-gwas-model.html#cb44-22" tabindex="-1"></a>geno_imputed <span class="ot">&lt;-</span> geno_imputed <span class="sc">%&gt;%</span></span>
<span id="cb44-23"><a href="how-do-you-include-pca-covariates-in-a-gwas-model.html#cb44-23" tabindex="-1"></a>  <span class="fu">rename</span>(<span class="at">FID =</span> <span class="dv">1</span>, <span class="at">IID =</span> <span class="dv">2</span>)</span>
<span id="cb44-24"><a href="how-do-you-include-pca-covariates-in-a-gwas-model.html#cb44-24" tabindex="-1"></a></span>
<span id="cb44-25"><a href="how-do-you-include-pca-covariates-in-a-gwas-model.html#cb44-25" tabindex="-1"></a><span class="co"># Step 5: Merge genotype with metadata</span></span>
<span id="cb44-26"><a href="how-do-you-include-pca-covariates-in-a-gwas-model.html#cb44-26" tabindex="-1"></a>geno_data <span class="ot">&lt;-</span> geno_imputed[, <span class="sc">-</span><span class="dv">1</span>]  <span class="co"># Drop FID, keep IID and SNPs</span></span>
<span id="cb44-27"><a href="how-do-you-include-pca-covariates-in-a-gwas-model.html#cb44-27" tabindex="-1"></a>gwas_input <span class="ot">&lt;-</span> <span class="fu">left_join</span>(sample_data, geno_data, <span class="at">by =</span> <span class="st">&quot;IID&quot;</span>)</span>
<span id="cb44-28"><a href="how-do-you-include-pca-covariates-in-a-gwas-model.html#cb44-28" tabindex="-1"></a></span>
<span id="cb44-29"><a href="how-do-you-include-pca-covariates-in-a-gwas-model.html#cb44-29" tabindex="-1"></a><span class="co"># Step 6: Select trait and covariates</span></span>
<span id="cb44-30"><a href="how-do-you-include-pca-covariates-in-a-gwas-model.html#cb44-30" tabindex="-1"></a>trait <span class="ot">&lt;-</span> <span class="st">&quot;Plant height&quot;</span>  <span class="co"># Use the column name as a string</span></span>
<span id="cb44-31"><a href="how-do-you-include-pca-covariates-in-a-gwas-model.html#cb44-31" tabindex="-1"></a>covariates <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;PC1&quot;</span>, <span class="st">&quot;PC2&quot;</span>, <span class="st">&quot;PC3&quot;</span>)</span>
<span id="cb44-32"><a href="how-do-you-include-pca-covariates-in-a-gwas-model.html#cb44-32" tabindex="-1"></a></span>
<span id="cb44-33"><a href="how-do-you-include-pca-covariates-in-a-gwas-model.html#cb44-33" tabindex="-1"></a><span class="co"># Step 7: Construct the model formula</span></span>
<span id="cb44-34"><a href="how-do-you-include-pca-covariates-in-a-gwas-model.html#cb44-34" tabindex="-1"></a>snp_name <span class="ot">&lt;-</span> <span class="fu">names</span>(geno_data)[<span class="dv">2</span>]  <span class="co"># Replace with desired SNP</span></span>
<span id="cb44-35"><a href="how-do-you-include-pca-covariates-in-a-gwas-model.html#cb44-35" tabindex="-1"></a>formula_str <span class="ot">&lt;-</span> <span class="fu">paste0</span>(<span class="st">&quot;`&quot;</span>, trait, <span class="st">&quot;` ~ &quot;</span>, snp_name, <span class="st">&quot; + &quot;</span>, <span class="fu">paste</span>(covariates, <span class="at">collapse =</span> <span class="st">&quot; + &quot;</span>))</span>
<span id="cb44-36"><a href="how-do-you-include-pca-covariates-in-a-gwas-model.html#cb44-36" tabindex="-1"></a>model <span class="ot">&lt;-</span> <span class="fu">lm</span>(<span class="fu">as.formula</span>(formula_str), <span class="at">data =</span> gwas_input)</span>
<span id="cb44-37"><a href="how-do-you-include-pca-covariates-in-a-gwas-model.html#cb44-37" tabindex="-1"></a></span>
<span id="cb44-38"><a href="how-do-you-include-pca-covariates-in-a-gwas-model.html#cb44-38" tabindex="-1"></a><span class="co"># Step 8: View model summary</span></span>
<span id="cb44-39"><a href="how-do-you-include-pca-covariates-in-a-gwas-model.html#cb44-39" tabindex="-1"></a><span class="fu">summary</span>(model)</span></code></pre></div>
<pre><code>
Call:
lm(formula = as.formula(formula_str), data = gwas_input)

Residuals:
    Min      1Q  Median      3Q     Max 
-54.024 -12.429  -0.346  11.594  76.555 

Coefficients:
             Estimate Std. Error t value Pr(&gt;|t|)    
(Intercept) 115.83448    1.04707 110.627  &lt; 2e-16 ***
id1000007     2.17767    1.54100   1.413    0.158    
PC1           0.20502    0.02761   7.426 7.49e-13 ***
PC2          -0.19534    0.04436  -4.404 1.39e-05 ***
PC3          -0.29738    0.07399  -4.019 7.05e-05 ***
---
Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1

Residual standard error: 18.63 on 378 degrees of freedom
  (30 observations deleted due to missingness)
Multiple R-squared:  0.2277,    Adjusted R-squared:  0.2195 
F-statistic: 27.86 on 4 and 378 DF,  p-value: &lt; 2.2e-16</code></pre>
<blockquote>
<p>‚úÖ <strong>Takeaway:</strong> Always ensure PCA scores include correctly labeled <code>FID</code> and <code>IID</code> so they can be merged with metadata and genotype matrices before fitting GWAS models.</p>
</blockquote>
</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="how-do-you-perform-pca-on-genotype-data-to-assess-population-structure.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="how-do-you-interpret-gwas-model-results-with-pca-covariates.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
  "sharing": {
    "github": false,
    "facebook": true,
    "twitter": true,
    "linkedin": false,
    "weibo": false,
    "instapaper": false,
    "vk": false,
    "whatsapp": false,
    "all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
  },
  "fontsettings": {
    "theme": "white",
    "family": "sans",
    "size": 2
  },
  "edit": {
    "link": "https://github.com/datainsights/cdi-gwas/edit/main/03-1-gwas.Rmd",
    "text": "Edit"
  },
  "history": {
    "link": null,
    "text": null
  },
  "view": {
    "link": null,
    "text": null
  },
  "download": ["gwas.pdf", "gwas.epub"],
  "search": {
    "engine": "fuse",
    "options": null
  },
  "toc": {
    "collapse": "subsection"
  }
});
});
</script>

</body>

</html>
